transform/transform_test.go

```diff
@@ -70,8 +70,10 @@ func fuzzyEqualIR(s1, s2 string) bool {
 func filterIrrelevantIRLines(lines []string) []string {
 	var out []string
 	for _, line := range lines {
-		line = strings.TrimSpace(line) // drop '\r' on Windows
-		if line == "" || line[0] == ';' {
+		line = strings.TrimSuffix(line, "\r") // drop '\r' on Windows
+		line = strings.Split(line, ";")[0]    // strip out comments/info
+		line = strings.TrimRight(line, " ")   // remove trailing spaces from comments
```

Comment:
Why is this necessary? It seems to be unnecessary for the PR here.

The new code appears to catch comments at the end of a line, but I think that can be done more cleanly this way:

```go
line = strings.Split(line, ";")[0] // strip out comments/info
line = strings.TrimSpace(line)     // drop '\r' on Windows and remove trailing spaces from comments
```

Reply:
I took out the `strings.TrimSpace` because it also strips leading spaces, which is actually incorrect because it strips out the indentation.

Reply:
I took out the `strings.TrimSpace` because it also strips leading spaces, which is actually incorrect because it strips out the indentation. This is more precise.
LLVM generates comments at the end of label lines.

Reply:
Indentation is not semantically relevant: you could use any kind of indentation in IR files.
The intention of this function was to compare whether two pieces of IR have any actual differences (excluding comments, whitespace, and other irrelevant details).

Reply:
>  LLVM generates comments at the end of label lines.

Ah, right. Good point.